Automattic/wp-calypso
# Add transfer status and logstash to debug woocommerce setup

#### Changes proposed in this Pull Request

This PR:
- Adds logstash to obtain more information (complete component state, transfer status, detailed list of plugins used in site) on WooCommerce setup during timeouts.
- Increases the timeout threshold for plugin installation and activation to 30 seconds. This is due to suspicion that sometimes plugin installation and activation might take more than 15 seconds.

#### Testing instructions

1. Run calypso in your development environment.
2. Manually induce stalling:
    - Open up `client/my-sites/woocommerce/dashboard/required-plugins-install-view.jsx`
    - Search for `this.props.installAndActivatePlugin( site.ID, plugin );`.
    - Comment out the line and wait for calypso to rebuild.
4. Create a new business site.
5. Install `Hello Dolly` plugin to turn it into Atomic site.
6. Navigate to `/woocommerce-installation/{your new site}`.
7. Click on `Set up my store!`.
8. Wait for 30 seconds, and observe "contact support" screen appears.
9. Open logstash and search for `feature: calypso_client and message: "woocommerce setup timeout"`
10. Observe that in `extra` key, `state, automatedTransferStatus, plugins, pluginsStatus`exists.
